rasa/utils/tensorflow/models.py

```diff
@@ -441,7 +443,7 @@ def load(
 
     @staticmethod
     def batch_to_model_data_format(
-        batch: Union[Tuple[tf.Tensor, ...], Tuple[np.ndarray, ...]],
+        batch: MaybeNestedBatchData,
```

Comment:
The type annotations here were imprecise since they never covered the case that the batch might be yet again nested in a tuple. All the methods leading to this function have still the old annotation, but it was never noticed since they just pass the data along and don't use it. I think these annotations should also be changed, but it's a lot of methods leading back to the batches coming out of the keras data iterator. So fixing those should be a different issue I believe

Reply:
Gotcha. I agree that fixing the entire chain is a separate issue - if you think it is important, could you create a follow up issue for it?